src/libraries/System.Text.Json/src/System/Text/Json/Serialization/Converters/Object/JsonObjectConverter.cs

```diff
@@ -9,7 +9,18 @@ namespace System.Text.Json.Serialization
     /// </summary>
     internal abstract class JsonObjectConverter<T> : JsonResumableConverter<T>
     {
+        internal JsonObjectConverter()
+        {
+            // Populate ElementType if the runtime type implements IAsyncEnumerable.
+            // Used to feed the (converter-agnostic) JsonClassInfo.ElementType instace
+            // which is subsequently consulted by custom enumerable converters fed via JsonConverterAttribute.
+            if (IAsyncEnumerableConverterFactory.TryGetAsyncEnumerableInterface(typeof(T), out Type? asyncEnumerableInterface))
```

Comment:
> Perhaps have the implementation of TryGetAsyncEnumerableInterface just check for the IAE interface and not loop though all interface implementations?

Do you mean we should restrict the feature to just `IAsyncEnumerable<T>` types instead of `TValue : IAsyncEnumerable<T>` types? Or is there a simpler way to check for the latter that doesn't involve looping?

Reply:
Could the `IAsyncEnumerableConverterFactory` be responsible for creating an instance of this converter, and setting the `ElementType` directly? (e.g through a new parameterized ctor)

Reply:
Per my comment in https://github.com/dotnet/runtime/pull/50778#discussion_r608838993, this can probably be removed altogether.